pkg/network/multus/annotations.go

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package multus
+
+import (
+	"fmt"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
+func IterateMultusNonDefaultNetworks(vmi *v1.VirtualMachineInstance, fn func(network v1.Network, networkRequestInterfaceName string)) {
+	multusNonDefaultNetworks := filterMultusNonDefaultNetworks(vmi.Spec.Networks)
+	for i, network := range multusNonDefaultNetworks {
+		networkRequestInterfaceName := fmt.Sprintf("net%d", i+1)
+		fn(network, networkRequestInterfaceName)
+	}
+}
```

Comment:
This function is wrong imo; it is basically a poor implementation of a map function that on top of that performs extra operations - has an embedded naming scheme, plus filters the multus non-default network.

I understand why you do it; you're being cleaver - unfortunately, that costs readability, something which I would rather maximize.

Furthermore, I see that one of the callers actually rely on this function to extract data, using a side-effect. 

I would:
- explicitly have a function that exposes the interface naming scheme based on the interface index. Make that usable by whoever needs it.
- have 2 functions that receive a list of networks - you'd pass the list of already filtered networks from the caller - then let each of the properly named functions iterate the list, call the naming scheme, and do what they need to do.

Reply:
I agree with the action items @maiqueb suggested.

Reply:
I added an interfaces that does just that. 
regarding the 2 functions - not sure I fully understood, but I think the change I inserted should answer all our needs